### ISSUE TITLE
Block Editor: Accessibility issue editing links

### ISSUE THREAD

**Issue by melindahelt** [wordpress-mobile/WordPress-iOS#14071]
Accessibility issue reported by a user:

#2937683-zen

hello, I have a problem on WordPress iOS app; I am a blind user, and I have set my iphone with its accessibility feature. Settings -> accessibility -> voiceover. Well, I have encountered a bug in your app, envolving VoiceOver usage; it occurs when you have placed links inside an article and have saved it into a draft.
Step-by-step:
1. Write an article including a link, using the "paragraph" block. The draft may be created using desktop platform.
2. Save the draft.
3. Open it with iOS app, and VoiceOver on.
4. Place VoiceOver focus on the block you wish to modify, where the link is.
5. Double-tap on the block...

The app will open the link, youtube or safari whatever it opens... And does not allow you to access keyboard to modify article text.
Only way to access editing function, is to press firmly on the screen using 3D touch, but soon 3D touch will disappear from iPhones...
Now I have no workaround for this, and I must remember to place a special marker in the words I want to link, then write it all, and place links as last element before publishing, on desktop. Please let me know!
Links should be opened only if I decide to do it, selecting "link" item from VoiceOver rotor.

##### Tested on [device], iOS [version], WPiOS [version]
iPhone X, iOS 13.4.5, WPiOS 14.6
